src/test/test-integration.ts

```diff
@@ -0,0 +1,95 @@
+
+/// <reference path="../../typings/knockout/knockout.d.ts" />
+/// <reference path="../../typings/jasmine/jasmine.d.ts" />
+/// <reference path="../../typings/socket.io-client/socket.io-client.d.ts" />
+
+import { App } from "../common/client-app";
+import * as viewModel from "../common/view-model";
+import * as model from "../common/model";
+import * as ko from "knockout";
+import { BoardVM } from "../koclient/ko-view-model";
+import {BoardService} from "../server/BoardService";
+
+describe("Integration test between client and service", function(){
+// TODO: maybe separate this two classes in a module
+
+  class ServerMock{
+  private _boardService: BoardService;
+  private _clients: Array<App>;
+  constructor(boardService:BoardService){
+    this._boardService = boardService;
+    this._clients = new Array<App>();
+    var server = this;
+    this._boardService.sendToClient = (changes) => {
+      for(var client in server._clients){
+        server._clients[client].onMessage({patch: changes});
+      }
+    };
+  }
+
+  public addClient = (client: App) => {
+    var server = this;
+    client.sendToServer = (changes) => {
+      server._boardService.onClientMessage(changes);
+    };
+    client.onMessage(this._boardService.onClientConnection());
+    this._clients.push(client);
+  }
+}
+
+  it("When a client are making a change, this changes shouldn't be applied on the same client", function(){
+      // Arrange
+        // Prepare variables
+      var boardVM: BoardVM = new BoardVM();
+      var client = new App(boardVM);
+      var noteVM = boardVM.createNote();
+      var beginingText = "This is the begining part of the test";
+      var endingText = " and this is the ending part of the test";
+      var boardService = new BoardService();
+      var serverMock = new ServerMock(boardService);
+      serverMock.addClient(client);
+
+      // Act
+      noteVM.content(beginingText);
+      client.onInterval();
+      noteVM.content(noteVM.content() + endingText);
+      boardService.onTic();
+
+      //Assert
+      expect(noteVM.content()).toEqual(beginingText + endingText);
```

Comment:
Nice!, I do not agree with the testcase description (_"changes shouldn't be applied"_), but I think that you are testing the right thing.

Now, I think that there could be _race conditions_ so, you should also tests other `client.onInterval()` / `boardService.onTic()` combinations.